power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -719,7 +722,7 @@ class TapPositionOptimizerImpl<std::tuple<TransformerTypes...>, StateCalculator,
         }
 
         void propose_new_pos(bool strategy_max, bool above_range) {
-            bool const is_down = above_range == tap_reverse_;
+            bool const is_down = (above_range == tap_reverse_) ^ control_at_tap_side_;
```

Comment:
please use `!=` instead of bitwise XOR for boolean values.

Rationale: the exact implementation of `bool` is not not specified in the C/C++ standard. As a result, bitwise operations on `bool`s are undefined behavior.

Example: if you have a `bool` variable represented by an `int8_t` and its binary representation is `00000001` another `bool` with a binary representation `00000010`, then their boolean values are both `1`. The boolean XOR should therefore be `0`. However, the bitwise XOR is `00000011`, which represents `1`, not `0`.